apps/frontend-pwa/src/components/learningElements/LearningElement.tsx

```diff
@@ -19,6 +20,9 @@ function LearningElement({
   handleNextQuestion,
 }: LearningElementProps) {
   const currentStack = element.stacks?.[currentIx]
+  const [stepStatus, setStepStatus] = useState<
+    ('unanswered' | 'incorrect' | 'partial' | 'correct')[]
+  >(Array.from({ length: element.stacks?.length ?? 0 }, () => 'unanswered'))
```

Comment:
might be easier to read to use functions from ramda for such cases, e.g.,

```suggestion
  R.repeat('unanswered', element.stacks?.length ?? 0)
```

Reply:
good idea, changed it